File: app/src/main/java/com/splatool/ikastyle/Entity/MainName.java
package com.splatool.ikastyle.Entity;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;

import com.splatool.ikastyle.Common.Util;


@Entity(tableName = "MAST_MAIN_NAME", primaryKeys = {"id", "category_id", "language_code"})
public class MainName {
    @ColumnInfo(name = "id")
    public int id;

    @ColumnInfo(name = "category_id")
    public int categoryId;

    @ColumnInfo(name = "language_code")
    public int languageCode;

    @ColumnInfo(name = "name")
    @NonNull
    public String name;

    @NonNull
    public String getName(){
        return name;
    }

    public int getAbsoluteId(){
        return Util.getAbsoluteId(categoryId, id, 0);
    }
}

File: app/src/main/java/com/splatool/ikastyle/model/data/entity/MainName.kt
package com.splatool.ikastyle.model.data.entity

import androidx.annotation.NonNull
import androidx.room.*
import com.splatool.ikastyle.model.common.Util

@Entity(tableName = "MAST_MAIN_NAME", primaryKeys = ["id", "category_id", "language_code"])
data class MainName (
    @ColumnInfo(name = "id") val id : Int,
    @ColumnInfo(name = "category_id") val categoryId : Int,
    @ColumnInfo(name = "language_code") val languageCode : Int,
    @ColumnInfo(name = "name") @NonNull val name: String
)
{
    fun getAbsoluteId(): Int {
        return Util.getAbsoluteId(categoryId, id, 0)
    }
}